spec/mongoid/indexable/specification_spec.rb

```diff
@@ -80,25 +80,79 @@
 
   describe "#initialize" do
 
-    let(:spec) do
-      described_class.new(
-        Band,
-        { name: 1, title: 1, years: -1 },
-        background: true,
-        unique: true
-      )
-    end
+    context "standard case" do
 
-    it "sets the class" do
-      expect(spec.klass).to eq(Band)
-    end
+      let(:spec) do
+        described_class.new(
+          Band,
+          { name: 1, title: 1, years: -1 },
+          background: true,
+          unique: true
+        )
+      end
+
+      it "sets the class" do
+        expect(spec.klass).to eq(Band)
+      end
 
-    it "normalizes the key" do
-      expect(spec.key).to eq(name: 1, title: 1, y: -1)
+      it "normalizes the key" do
+        expect(spec.key).to eq(name: 1, title: 1, y: -1)
+      end
+
+      it "normalizes the options" do
+        expect(spec.options).to eq(background: true, unique: true)
+      end
     end
 
-    it "normalizes the options" do
-      expect(spec.options).to eq(background: true, unique: true)
+    context "with aliased field options" do
```

Comment:
Can you add a _bunch_ more tests than just the one you have here? Some thoughts:

1. What happens when it's not under one of the three keys
2. What happens when we have an arbitrary $
3. What happens when giving a hash of arbitrary depth

Try to also make separate tests with the different keys so that we're not testing it all at once (you can keep the one with all 3 just add more) this way if any of them break we'll know which easily...

Reply:
@Neilshweky I've taken a second look at this. When considering this spec plus the many specs in spec/mongoid/indexable_spec.rb (below) which are testing the same thing, I really don't think there are more cases to add, I've covered all the canonical "real" cases and a good cross section of the possible non-sense.

If you can give me a list of specific extra scenarios you have in mind--and please double-check my specs don't cover them already--I'll be happy to add them. Otherwise, let's mark this done.